src/common/utils/collection-functions.ts

```diff
@@ -69,7 +69,8 @@ export async function getOrInsertWithAsync<K, V>(map: Map<K, V>, key: K, asyncBu
     map.set(key, await asyncBuilder());
   }
 
-  return map.get(key);
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  return map.get(key)!;
```

Comment:
Hmm. What's going on here? Is the new strictness causing complexity?

Reply:
Sort of, I decided to do it this way because getting once and checking if the value is `=== undefined` doesn't work in the generic case because `T` might actually be `string | undefined`.

Reply:
Hmm. Do you figure we will start seeing more escapes like this?

Reply:
I do not, one of the reasons why these functions are here. So that these escapes can be abstracted away.